ports/atomvm_mqtt_client.c

```diff
@@ -785,16 +791,23 @@ Context *atomvm_mqtt_client_create_port(GlobalContext *global, term opts)
     UNUSED(port);
     char *username_str = maybe_get_string(opts, username_atom, global);
     char *password_str = maybe_get_string(opts, password_atom, global);
-    // char *cert_str = maybe_get_string(opts, cert_atom, global);
+    char *client_id = maybe_get_string_or_default(opts, client_id_atom, get_default_client_id(), global);
```

Comment:
I think we should keep the naming scheme consistent and use `client_id_str` here, which would require updating lines 802 and 811 below.